app/parse.py

```
@@ -15,8 +32,128 @@ class Product:
     num_of_reviews: int
 
 
+def parse_product(element: WebElement) -> Product:
+    title = element.find_element(By.CLASS_NAME, "title").get_attribute("title")
+    description = element.find_element(
+        By.CLASS_NAME, "description"
+    ).text.strip()
+
+    price = float(
+        element.find_element(By.CLASS_NAME, "price").text.replace("$", "")
+    )
+    rating = len(
+        element.find_elements(By.CSS_SELECTOR, ".ratings .ws-icon-star")
+    )
+    num_of_reviews = int(
+        element.find_element(By.CLASS_NAME, "review-count")
+        .text.strip()
+        .split()[0]
+    )
+
+    return Product(title, description, price, rating, num_of_reviews)
+
+
+def parse_page(url: str) -> list[Product]:
+    driver.get(url)
+    products = []
+
+    try:
+        accept_btn = WebDriverWait(driver, 1).until(
+            expected_conditions.element_to_be_clickable(
+                (
+                    By.XPATH,
+                    "//button[contains(., 'Accept') or contains(., 'OK')]",
+                )
+            )
+        )
+        accept_btn.click()
+    except Exception:
+        pass
```

Cookie consent handling is fragile: the wait is only 1 second and the code catches a broad `Exception`. Increase the wait (≈1–2s) and catch a specific exception such as `TimeoutException` so unrelated errors are not swallowed.